pkg/proxy/proxy_test.go

```diff
@@ -0,0 +1,104 @@
+	p.invokeGrpc(httptest.NewRecorder(), request)
+
+	inputSignals := inputSignals(invokeClient.Calls)
+	startFrame := inputSignals[0].GetStart()
+	assert.Equal(t, []string{"text/plain+accept"}, startFrame.ExpectedContentTypes)
+}
+
+func Test_invokeGrpc_input_dataFrame(t *testing.T) {
+	riffClient, invokeClient := mockRiffClient()
+	p := &proxy{riffClient: riffClient}
+
+	request, _ := http.NewRequest("POST", "/invoker-url", strings.NewReader("some body"))
+	request.Header.Add("content-type", "text/plain+content-type")
+	request.Header.Add("x-custom-header", "header-value")
+	p.invokeGrpc(httptest.NewRecorder(), request)
+
+	inputSignals := inputSignals(invokeClient.Calls)
+	dataFrame := inputSignals[1].GetData()
+	assert.Equal(t, "some body", string(dataFrame.Payload))
+	assert.Equal(t, "text/plain+content-type", dataFrame.ContentType)
+	assert.Contains(t, dataFrame.Headers, "X-Custom-Header")
+	assert.Equal(t, dataFrame.Headers["X-Custom-Header"], "header-value")
+}
+
+func Test_invokeGrpc_output(t *testing.T) {
+	riffClient, _ := mockRiffClientWithResponse("some response")
+	p := &proxy{riffClient: riffClient}
+
+	request, _ := http.NewRequest("POST", "/invoker-url", strings.NewReader(""))
+	responseRecorder := httptest.NewRecorder()
+	p.invokeGrpc(responseRecorder, request)
+
+	assert.Equal(t, "some response", responseRecorder.Body.String())
+	assert.Equal(t, "text/plain+response", responseRecorder.Header().Get("Content-Type"))
+}
+
+func Test_invokeGrpc_wiring(t *testing.T) {
+	riffClient, invokeClient := mockRiffClient()
+	p := &proxy{riffClient: riffClient}
+
+	request, _ := http.NewRequest("POST", "/invoker-url", strings.NewReader("some body"))
+	p.invokeGrpc(httptest.NewRecorder(), request)
+
+	riffClient.AssertExpectations(t)
+	invokeClient.AssertExpectations(t)
+}
+
+func inputSignals(calls []mock.Call) []*rpc.InputSignal {
+	var inputSignals []*rpc.InputSignal
+	for _, call := range calls {
+		if call.Method == "Send" {
+			signal := call.Arguments.Get(0).(*rpc.InputSignal)
+			inputSignals = append(inputSignals, signal)
+		}
+	}
+	return inputSignals
+}
+
+func mockRiffClient() (*mocks.RiffClient, *mocks.Riff_InvokeClient) {
+	return mockRiffClientWithResponse("")
+}
+
+func mockRiffClientWithResponse(outputBody string) (*mocks.RiffClient, *mocks.Riff_InvokeClient) {
+	riffClient := &mocks.RiffClient{}
+	invokeClient := &mocks.Riff_InvokeClient{}
+	riffClient.On("Invoke", context.Background()).Return(invokeClient, nil)
+	invokeClient.On("Send", mock.Anything).Return(nil)
+	invokeClient.On("CloseSend").Return(nil)
+	invokeClient.On("Recv").Return(outputSignal(outputBody), nil).Once()
+	invokeClient.On("Recv").Return(nil, io.EOF)
+	return riffClient, invokeClient
+}
+
+func outputSignal(outputBody string) *rpc.OutputSignal {
+	return &rpc.OutputSignal{
+		Frame: &rpc.OutputSignal_Data{
+			Data: &rpc.OutputFrame{
+				Payload:     []byte(outputBody),
+				ContentType: "text/plain+response",
```

Comment:
Could be made a parameter, as it's not always obvious at usage point why this particular content-type is being used